### ISSUE TITLE
Trades DB updates on next loop around pairs after close

### ISSUE THREAD

**Issue by creslinux** [freqtrade/freqtrade#1053]
The below excepts shows logging, and some prints, of a pair from many 

1) Shows the buy
2) Show the sell/close - and update the RPC with sale
3) Shows when the sell/close update trades DB 

In summary trades DB does not update when a pair is sold. 
The trades DB updates on the next loop which may be minutes later. 

```
==========

BUY LINK/BTC
buy id is dry_run_buy_267336
2018-07-21 10:34:32,662 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': buy, 'exchange': 'Binance', 'pair': 'LINK/BTC', 'market_url': 'https://www.binance.com/tradeDetail.html?symbol=LINK_BTC', 'limit': 2.745e-05, 'stake_amount': 0.05, 'stake_amount_fiat': 0, 'stake_currency': 'BTC', 'fiat_currency': 'USD'}

SELL LINK/BTC
---------------we are in strategy 10:35:15.896980
sell = 1
closed with  dry_run_sell_321851
2018-07-21 10:35:15,915 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=2, pair=LINK/BTC, amount=1821.49362477, open_rate=0.00002745, open_since=seconds ago)
2018-07-21 10:35:15,915 - freqtrade.freqtradebot - INFO - Applying fee on amount for Trade(id=2, pair=LINK/BTC, amount=1821.49362477, open_rate=0.00002745, open_since=seconds ago) failed: myTrade-Dict empty found
2018-07-21 10:35:15,915 - freqtrade.persistence - INFO - Updating trade (id=2) ...
2018-07-21 10:35:15,915 - freqtrade.persistence - INFO - LIMIT_BUY has been fulfilled for Trade(id=2, pair=LINK/BTC, amount=1821.49362477, open_rate=0.00002745, open_since=seconds ago).
conn is HTTPSConnectionPool(host='api.binance.com', port=443)
conn is HTTPSConnectionPool(host='api.binance.com', port=443)
----------------we are in strategy 10:35:16.896002

UPDATE TRADES TABLE for LINK/BTC 
now closing {'pair': 'LINK/BTC', 'price': 2.731e-05, 'amount': 1821.4936247723135, 'type': 'limit', 'side': 'sell', 'remaining': 0.0, 'datetime': '2018-07-21T10:35:16.899840+00:00', 'status': 'closed', 'id': 'dry_run_sell_839559'}
2018-07-21 10:36:07,575 - freqtrade.freqtradebot - INFO - Applying fee on amount for Trade(id=2, pair=LINK/BTC, amount=1821.49362477, open_rate=0.00002745, open_since=2 minutes ago) failed: myTrade-Dict empty found
2018-07-21 10:36:07,575 - freqtrade.persistence - INFO - Updating trade (id=2) ...
2018-07-21 10:36:07,575 - freqtrade.persistence - INFO - Marking Trade(id=2, pair=LINK/BTC, amount=1821.49362477, open_rate=0.00002745, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.

```

**Comment by xmatthias** [comment]
i think this is correct behavior / don't see the bug in it... 

it may look odd in dry-run, but it's creating the close order - and in the consequent loop checks if that order is still open (and if it's fullfilled, which in dry-run it will always be in the next loop, it updates the database with the "close" flag).
In dry-run, it could assume it's always immediate sell - but that's not a simulation as this cannot happen in live (we always need to create order, and then check if it went through).

In live, you could also have it run 2-3 loops (or more in a low volume market) until the sell Really happens - the bot should not mark the trade as closed until that point...

**Comment by xmatthias** [comment]
your video is missing one crucial information, which is "open_order_id", which i think is updated right after `exchange.sell` is called - and the bot has to give the exchange a bit of time (in real mode, anyway) to execute the sell. For dry-mode it will work fine to immediately ask and get "yes" ... but i doubt that works in production too, as the result can be "still open".

However - i think this all boils down to one root problem - which was outlined in #1046 - bot beeing slow.
If that is fixed, this issue should be reduced by a ton.

**Comment by xmatthias** [comment]
closed in #2124 (merge did fail - but it's merged (https://github.com/freqtrade/freqtrade/commit/3af5691b9186e70dc879d623d1520ff17393495c).
